compiler/rustc_hir_analysis/src/check/wfcheck.rs

```diff
@@ -1749,7 +1750,20 @@ fn check_method_receiver<'tcx>(
             {
                 match receiver_validity_err {
                     ReceiverValidityError::DoesNotDeref if arbitrary_self_types_level.is_some() => {
-                        tcx.dcx().emit_err(errors::InvalidReceiverTy { span, receiver_ty })
+                        let hint = match receiver_ty
+                            .ty_def_id()
```

Comment:
don't use this, it's an internal detail of the. use `Ty::ty_adt_def` to check that it's an adt.

Reply:
Thanks - will do